datasets/plugins/batch/batch_dataset_plugin.py

```diff
@@ -0,0 +1,173 @@
+from pathlib import Path
+from typing import TYPE_CHECKING, Iterable, Optional, Tuple, Union
+
+import pandas
+import pandas as pd
+
+from datasets import Mode
+from datasets.context import Context
+from datasets.dataset_plugin import DatasetPlugin
+
+
+if TYPE_CHECKING:
+    import dask.dataframe as dd
+    import pyspark
+
+
+class InvalidOperationException(Exception):
+    pass
+
+
+@DatasetPlugin.register_plugin(constructor_keys={"name"}, context=Context.BATCH)
+class BatchDatasetPlugin(DatasetPlugin):
+    """
+    This is the default plugin for the BATCH execution context.
+    """
+
+    _dataset_path_func: callable = None
+
+    def __init__(
+        self,
+        name: str,
+        logical_key: str = None,
+        columns: Optional[Union[Iterable[str], str]] = None,
+        run_id: Optional[str] = None,
+        mode: Mode = Mode.READ,
+        class_field_name: Optional[str] = None,
+        partition_by: Optional[str] = None,
+        path: Optional[str] = None,
+    ):
+        self.path = path
+        self.partition_by = partition_by
+        self.program_name = self._executor.current_program_name
+        super(BatchDatasetPlugin, self).__init__(
+            name=name,
+            logical_key=logical_key,
+            columns=columns,
+            run_id=run_id,
+            mode=mode,
+            class_field_name=class_field_name,
+        )
+
+    def _get_path_filters_columns(
+        self, columns: Optional[Union[Iterable[str], str]] = None, run_id: Optional[str] = None
+    ) -> Tuple[str, Optional[list], Optional[Iterable[str]]]:
+        path = self._get_dataset_path()
+        read_columns = self._get_read_columns(columns)
+        filters = None
+        query_run_id = run_id if run_id else self.run_id
+        if query_run_id:
+            filters = [("run_id", "=", query_run_id)]
+        return path, filters, read_columns
+
+    def read_pandas(
+        self,
+        columns: Optional[str] = None,
+        storage_format: str = "parquet",
+        run_id: Optional[str] = None,
+        **kwargs,
+    ) -> pd.DataFrame:
+        path, filters, read_columns = self._get_path_filters_columns(columns, run_id=run_id)
+
+        df: pd.DataFrame
+        if storage_format == "parquet":
+            df = pandas.read_parquet(path, columns=read_columns, engine="pyarrow", filters=filters, **kwargs)
```

Comment:
Do we want to expose the `engine` type as a configurable parameter? Obviously defaulting to `pyarrow`.